arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java

```diff
@@ -304,19 +302,7 @@ public void populateColumnsForPrecompile(MessageFrame frame) {
     final int cdsIndex = opCode.callCanTransferValue() ? 4 : 3;
     final int returnAtCapacityIndex = opCode.callCanTransferValue() ? 6 : 5;
 
-    final boolean transfersValue = opCode.callCanTransferValue() && !frame.getStackItem(2).isZero();
-
-    // shameless copy from gasAvailableForChildCall found in TangerineWhistleGasCalculator
-    // TODO: @Olivier and @François: find out whether frame.getRemainingGas was already
-    //  decremented by the upfront cost. If not we must replace remainingGas with the
-    //  decremented version
-    long remainingGas = frame.getRemainingGas();
-    long gasCap =
-        Words.unsignedMin(
-            allButOneSixtyFourth(remainingGas), Words.clampedToLong(frame.getStackItem(0)));
-    long callGasLong =
-        transfersValue ? gasCap + ZkTracer.gasCalculator.getAdditionalCallStipend() : gasCap;
-    final BigInteger callGas = BigInteger.valueOf(callGasLong);
+    final BigInteger calleeGas = ((PrecompileCommonOobCall) oobCall).getCalleeGas();
```

Comment:
This here is the point of everything else.

Reply:
We are no longer (falsely) recomputing the gas allowance of the callee. We are getting it like so

```rust
STP -> CallSection -> OobCall ->OobOperation
```